linera-core/src/chain_worker/actor.rs

```diff
@@ -163,7 +163,8 @@ where
     pub async fn load(
         config: ChainWorkerConfig,
         storage: StorageClient,
-        certificate_value_cache: Arc<ValueCache<CryptoHash, HashedCertificateValue>>,
+        confirmed_value_cache: Arc<ValueCache<CryptoHash, Hashed<ConfirmedBlock>>>,
+        validated_value_cache: Arc<ValueCache<CryptoHash, Hashed<ValidatedBlock>>>,
```

Comment:
(If I understand correctly, this could be unified again if we hashed only the `ExecutedBlock` and then signed the hash plus a discriminant?)

Reply:
Yes